### ISSUE TITLE
Add will-frame-navigate event

### ISSUE THREAD

**Issue by bdxing** [electron/electron#7097]
- Electron version:
## \* Operating system: window 7

Found in webContents listening 'will - navigate' with 'event callback function event. The preventDefault ();'can't stop jumping events in the iframe.Example: in the iframe "< a href =" github.com "> test < / a >"

**Comment by bontibon** [comment]
#1378

**Comment by kevinsawicki** [comment]
Closing as duplicate of #1378

**Comment by deepak1556** [comment]
I am not sure if this is a bug or enhancement, the original event was never specified for sub-frame navigations. Will defer to others for thoughts.

**Comment by MarshallOfSound** [comment]
@deepak1556 Should we just expose the `isMainFrame` property on the event like we do on things like `did-navigate-in-page` so used can filter out iframe events if they want to?

**Comment by bdxing** [comment]
@deepak1556 @MarshallOfSound @kevinsawicki  `did-navigate-in-page` Don't listen to change the url in the iframe, tested

**Comment by MarshallOfSound** [comment]
@bdxing  `did-navigate-in-page` definitely fires in iFrames as I fixed an issue related to it #6808

**Comment by zcbenz** [comment]
`did-navigate-in-page` only emits for in-page navigations, that's probably why it doesn't emit for you.

We need to add a new `will-frame-navigate` event that also emits for subframe navigations, changing the behavior of `will-navigate` is not an option, since most code using the event assumes it only emits for the main frame.

**Comment by bdxing** [comment]
@zcbenz OK, The first thank you, waiting for the event.

**Comment by burtonator** [comment]
Wow.. I actually didn't realize this was missing and probably opened up a security hole if someone really wanted to exploit it.  If you assume you're not allowing iframes to load by preventing them then you could accidentally load a page when you intended to abort.

**Comment by miniak** [comment]
cc @samuelmaddock

**Comment by samuelmaddock** [comment]
This functionality should be easy enough to add after #29290 is merged.

**Comment by pushkin-** [comment]
@samuelmaddock It would be nice to add this now that the other PR got merged if someone on the team has some time :)

**Comment by HimanshuNarware** [comment]
hey, I want to work on this issue,please assing me this issue.

**Comment by codebytere** [comment]
@HimanshuNarware this issue is already in progress and has an assignee and PR. You're welcome to try and find another if you wish!
